core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -149,6 +149,10 @@ public BaseMetastoreCatalogTableBuilder(TableIdentifier identifier, Schema schem
       this.identifier = identifier;
       this.schema = schema;
       this.tableProperties.putAll(tableDefaultProperties());
+      // Explicitly set ZSTD for new tables
+      this.tableProperties.put(
+          TableProperties.PARQUET_COMPRESSION,
+          TableProperties.PARQUET_COMPRESSION_NEW_TABLE_DEFAULT);
```

Comment:
```Java
this.tableProperties.put(
    TableProperties.PARQUET_COMPRESSION,
    TableProperties.PARQUET_COMPRESSION_DEFAULT);
```
so we don't need to introduce a new conf, `PARQUET_COMPRESSION_NEW_TABLE_DEFAULT`